### PR TITLE
Unit Tests : adding tests for VDimension, VMeasure, Domain, ExtendedChoiceFormat and getSearchTable

### PR DESCRIPTION
**Pull request by MedAliMarz**
This PR adds unit tasks for the classes:
- VDimension
- VMeasure
- Domain
- ExtendedChoiceFormat

and for the function: getSearchTable (with corner cases)

### CODE REVIEW

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt [hunk 1]
```diff
@@ -198,4 +198,124 @@ class FormWithNullableColumnsTest : JApplicationTestBase() {
...
+      assertEquals("SELECT CLIENTS.ID, ADRESS.ID, PRODUCTS.ID, CLIENTS.\"NAME\", CLIENTS.MAIL," +
+              " \"ORDER\".QUANTITY FROM \"ORDER\" LEFT JOIN CLIENTS " +
+              "ON \"ORDER\".CLIENT_ID = CLIENTS.ID INNER JOIN ADRESS " +
+              "ON \"ORDER\".ADDRESS_ID = ADRESS.ID LEFT JOIN PRODUCTS " +
+              "ON \"ORDER\".PRODUCT_ID = PRODUCTS.ID", query.prepareSQL(this) )
+    }
+  }
+
+  @Test
+  fun InnerLeftInnerJoinFourTables() {
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt:225]: typo : *inner

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt [hunk 2]
```diff
@@ -198,4 +198,124 @@ class FormWithNullableColumnsTest : JApplicationTestBase() {
...
+      assertEquals("SELECT ADRESS.ID, CLIENTS.ID, \"ORDER\".PRODUCT_ID, CLIENTS.\"NAME\", CLIENTS.MAIL," +
+              " \"ORDER\".QUANTITY FROM ADRESS INNER JOIN \"ORDER\" " +
+              "ON ADRESS.ID = \"ORDER\".ADDRESS_ID LEFT JOIN CLIENTS " +
+              "ON CLIENTS.ID = \"ORDER\".CLIENT_ID INNER JOIN PRODUCTS " +
+              "ON PRODUCTS.ID = \"ORDER\".PRODUCT_ID" , query.prepareSQL(this) )
+    }
+  }
+
+  @Test
+  fun InnerJoinNoJoinTwoTables() {
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt:247]: typo : *inner

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt [hunk 3]
```diff
@@ -198,4 +198,124 @@ class FormWithNullableColumnsTest : JApplicationTestBase() {
...
+
+    transaction {
+      assertEquals("SELECT ADRESS.ID, \"ORDER\".CLIENT_ID " +
+              "FROM ADRESS INNER JOIN \"ORDER\" " +
+              "ON ADRESS.ID = \"ORDER\".ADDRESS_ID", query.prepareSQL(this) )
+    }
+  }
+
+  @Test
+  fun NoJoinInnerJoinTwoTables() {
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt:266]: typo : *noJoin

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt [hunk 4]
```diff
@@ -198,4 +198,124 @@ class FormWithNullableColumnsTest : JApplicationTestBase() {
...
+
+    transaction {
+      assertEquals("SELECT \"ORDER\".CLIENT_ID, ADRESS.ID "  +
+              "FROM ADRESS INNER JOIN \"ORDER\" " +
+              "ON ADRESS.ID = \"ORDER\".ADDRESS_ID", query.prepareSQL(this) )
+    }
+  }
+
+  @Test
+  fun LeftJoinNoJoinTwoTables() {
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumnTest.kt:285]: typo : *leftJoin

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumns.kt [hunk 5]
```diff
@@ -268,6 +278,96 @@ object FormWithNullableColumn : DictionaryForm() {
       }
     }
   }
+
+  val blockFourTablesLeftInnerLeftJoin = p1.insertBlock(FormFourTablesLeftInnerLeftJoin()) {
+    command(item = FormWithList.list) {
+      action = {
+        println("-----------Generating list-----------------")
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumns.kt:285]: delete all println

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumns.kt [hunk 6]
```diff
@@ -548,6 +648,110 @@ class FormThreeTablesEndLeftJoinInField : FormBlock(1, 1, "Left Join in the end
...
+
+class FormTwoTablesInnerJoinNoJoin : FormBlock(1, 1, "Inner join then no-join Test") {
+  val a = table(Adress)
+  val o = table(Order)
+
+  val id_address = skipped(domain = INT(20), position = at(2, 1)) {
+    columns(a.id, o.address_id)
+  }
+  val id_user = skipped(domain = INT(20), position = at(1, 1)) {
+    columns(o.client_id)
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithNullableColumns.kt:714]: when you put all field as skipped you can't access to this block

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/base/ExtendedChoiceFormatTests.kt [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    val extendedChoiceFormat = ExtendedChoiceFormat(limits= doubleArrayOf(0.0,0.5,1.0), arrayOf<String>("zero", "half", "one"), true)
+
+    @Test
+    fun testFormat(){
+        assertEquals("one", extendedChoiceFormat.format(1.0))
+        assertEquals("one", extendedChoiceFormat.format(1))
+        assertEquals("half", extendedChoiceFormat.format(0.5))
+        assertEquals("zero", extendedChoiceFormat.format(0))
+        assertEquals("zero", extendedChoiceFormat.format(0.0))
```
[the remark this answers is not in the document]
Comment (MedAliMarz) [reply galite-tests/src/test/kotlin/org/kopi/galite/tests/base/ExtendedChoiceFormatTests.kt:34]: Done

**Review by h-haddad — CHANGES_REQUESTED** [review]

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/chart/VMeasureTests.kt [hunk 8]
```diff
@@ -17,96 +17,110 @@
...
+import org.kopi.galite.visual.domain.INT
+import org.kopi.galite.visual.domain.STRING
 import org.kopi.galite.visual.dsl.chart.Chart
 import org.kopi.galite.visual.type.Decimal
-import org.kopi.galite.visual.visual.Color
-import org.kopi.galite.visual.visual.VColor
-import java.util.*
-import kotlin.test.assertEquals
 
 class VMeasureTests {
```
[the remark this answers is not in the document]
Comment (MedAliMarz) [reply galite-tests/src/test/kotlin/org/kopi/galite/tests/chart/VMeasureTests.kt:36]: As we are testing the VMeasure class, we don't have any functionalities to verify, therefore we check if the attributes are passed correctly to the internal model

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/chart/VMeasureTests.kt [hunk 9]
```diff
@@ -17,96 +17,110 @@
...
+      this[integerCodeMeasure] = 1
     }
-}
\ No newline at end of file
+
+    decimalDimension.add("First Decimal") {
+      this[decimalMeasure] = Decimal(15.0)
+      this[decimalCodeMeasure] = Decimal(0.0)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (MedAliMarz) [reply galite-tests/src/test/kotlin/org/kopi/galite/tests/chart/VMeasureTests.kt]: Done

**Review by h-haddad — APPROVED** [review]

**Review by hfazai — APPROVED** [review]
